common/data_refinery_common/models/ontology_term.py

```diff
@@ -0,0 +1,116 @@
+    # it doesn't matter which database we get it from
+    for term in terms:
+        # Sanity check if for some reason the first item returned isn't actually the one we want
+        if term["obo_id"] == ontology_term:
+            return term["label"]
+
+    raise ValueError("We can't find {} in the Ontology Lookup Service".format(ontology_term))
+
+
+class OntologyTerm(models.Model):
+    """ The mapping between a human-readable name and an ontology term """
+
+    class Meta:
+        db_table = "ontology_terms"
+
+    ontology_term = models.TextField(unique=True)
+    human_readable_name = models.TextField()
+
+    def to_dict(self):
+        return {
+            "term": self.ontology_term,
+            "name": self.human_readable_name,
+        }
+
+    @staticmethod
+    def _get_ontology_prefix(ontology_term: str) -> str:
+        """ Returns the ontology prefix for an ontology term
+
+        Example: For EFO:0002939, it would return efo
+        """
+        return ontology_term.split(":")[0].lower()
+
+    @staticmethod
+    def import_entire_ontology(ontology_prefix: str):
+        """ Given an ontology prefix, download the entire ontology and import it """
+        response = requests.get(ONTOLOGY_URL_TEMPLATE.format(ontology_prefix.lower()))
+        ontology_xml = ET.fromstring(response.text)
+
+        namespace = {
+            "owl": "http://www.w3.org/2002/07/owl#",
+            "rdf": "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
+            "rdfs": "http://www.w3.org/2000/01/rdf-schema#",
+        }
+
+        # For some reason, there are two ways to find ontology terms in OWL files.
+        # The first is <owl:Class> tags with an <rdfs:label> child
+        for child in ontology_xml.findall("owl:Class/[rdfs:label]", namespace):
+            about = child.attrib.get("{" + namespace["rdf"] + "}about")
+            ontology_term = about.split("/")[-1].replace("_", ":")
+            human_readable_name = child.find("rdfs:label", namespace).text
+
+            if OntologyTerm._get_ontology_prefix(ontology_term) == ontology_prefix:
+                term, _ = OntologyTerm.objects.get_or_create(ontology_term=ontology_term)
+                term.human_readable_name = human_readable_name
+                term.save()
+
+        # The her way is <rdf:Description> tags with an <rdfs:label> child
+        for child in ontology_xml.findall("rdf:Description/[rdfs:label]", namespace):
+            about = child.attrib.get("{" + namespace["rdf"] + "}about")
+            ontology_term = about.split("/")[-1].replace("_", ":")
+            human_readable_name = child.find("rdfs:label", namespace).text
+
+            if OntologyTerm._get_ontology_prefix(ontology_term) == ontology_prefix:
+                term, _ = OntologyTerm.objects.get_or_create(ontology_term=ontology_term)
+                term.human_readable_name = human_readable_name
+                term.save()
+
+    @classmethod
+    def poke_term(cls, ontology_term: str):
```

Comment:
This method is basically identicall to `get_or_create_from_api`. Can that just be called instead of this and just ignore its return value?

Reply:
Good point